api/src/models/user.rs

```diff
@@ -30,9 +31,10 @@ impl User {
     }
 }
 
-#[derive(Debug, Default, PartialEq, Insertable)]
+#[derive(Debug, Default, PartialEq, Insertable, Validate)]
 #[table_name = "users"]
 pub struct NewUser<'a> {
+    #[validate(length(min = 1))]
```

Comment:
Should this also have the max len of 20 validate or should we just make username be `TEXT` instead of varchar?

Reply:
That's a good point. I'm still planning on switch to just using the DB constraints for validation at some point, but for now since we're still using `validator` I'll add that in